src/datarobotx/idp/custom_application_source.py

```diff
@@ -0,0 +1,316 @@
+    previous_custom_application_source_version_id: Optional[str] = None,
+    **kwargs: Any,
+) -> str:
+    """
+    Get or create a custom application source version.
+
+    Parameters
+    ----------
+    custom_application_source_id : str
+        The ID of the custom application source.
+    runtime_parameter_values : Optional[List[Union[RuntimeParameterValue, Dict[str, str]]]], optional
+        A list of runtime parameter values. Each value can be either a `RuntimeParameterValue` object or a dictionary
+    previous_custom_application_source_version_id : Optional[str], optional
+        The ID of the previous custom application source version. (default: None) If provided,
+        the version will be changed in place, no new version will be created
+    **kwargs : Any
+        Additional keyword arguments.
+
+    Returns
+    -------
+    str
+        The ID of the custom application source version.
+    """
+    return _get_or_create_custom_application_source_version(
+        endpoint=endpoint,
+        token=token,
+        custom_application_source_id=custom_application_source_id,
+        runtime_parameter_values=runtime_parameter_values,
+        previous_custom_application_source_version_id=previous_custom_application_source_version_id,
+        **kwargs,
+    )
+
+
+# def get_or_create_custom_application_source_version_from_previous(
+#     endpoint: str,
+#     token: str,
+#     custom_application_source_id: str,
+#     previous_custom_application_source_version_id: str,
+#     runtime_parameter_values: Optional[List[Union[RuntimeParameterValue, Dict[str, str]]]] = None,
+#     **kwargs: Any,
+# ) -> str:
+#     """
+#     Get or create a custom application source version from a previous version.
+
+#     Parameters
+#     ----------
+#     custom_application_source_id : str
+#         The ID of the custom application source.
+#     previous_custom_application_source_version_id : str
+#         The ID of the previous custom application source version.
+#     runtime_parameter_values : Optional[List[Union[RuntimeParameterValue, Dict[str, str]]]], optional
+#         A list of runtime parameter values. Each value can be either a `RuntimeParameterValue` object or a dictionary
+#         with parameter names as keys and parameter values as values. (default: None)
+#     **kwargs : Any
+#         Additional keyword arguments.
+
+#     Returns
+#     -------
+#     str
+#         The ID of the custom application source version.
+#     """
+#     return _get_or_create_custom_application_source_version(
+#         from_previous=True,
+#         endpoint=endpoint,
+#         token=token,
+#         custom_application_source_id=custom_application_source_id,
+#         previous_custom_application_source_version_id=previous_custom_application_source_version_id,
+#         runtime_parameter_values=runtime_parameter_values,
+#         **kwargs,
+#     )
+
+
+def get_or_create_qanda_app(
+    endpoint: Optional[str],
```

Comment:
we are moving away from altering the signature for endpoint/token unless we standardize everywhere, no?